core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -12648,6 +12648,11 @@ public User getUser(String userID, String userName) throws UserStoreException {
         if (userName == null) {
             userName = getUserNameFromUserID(userID);
         }
+
+        if (StringUtils.isEmpty(userID) || StringUtils.isEmpty(userName)) {
```

Comment:
This logic is improved by this PR - https://github.com/wso2/carbon-kernel/pull/3597